src/App/Fossa/ArchiveUploader.hs

```diff
@@ -84,6 +97,24 @@ archiveUploadSourceUnit baseDir apiOpts vendoredDeps = do
 archiveNoUploadSourceUnit :: [VendoredDependency] -> [Locator]
 archiveNoUploadSourceUnit = map (arcToLocator . forceVendoredToArchive)
 
+duplicateNames :: [VendoredDependency] -> [Text]
+duplicateNames deps = map extract $ filter isDuplicated (Map.toList . Map.fromListWith (+) . map pair $ deps)
```

Comment:
[tiniest nits of all time; this code is actually fine, so feel free to ignore]

- `fst :: (a,b) -> a` is the Prelude-builtin name for `extract`, but consider using `Map.keys`
- consider using `Map.filter`

```hs
duplicateNames deps = Map.keys . Map.filter (> 1) . Map.fromListWith (+) $ map pair deps
```

or for extra unnecessary type safety + breaking out smaller functions:

```hs
duplicateNames :: [VendoredDependency] -> [Text]
duplicateNames = duplicates . map vendoredName

duplicates :: Ord a => [a] -> [a]
duplicates = Map.keys . Map.filter (>1) . count

count :: Ord a => [a] -> Map a Int
count = Map.fromListWith (+) . map (\k -> (k,1))
```